.github/workflows/GPUCompat.yml

```diff
@@ -0,0 +1,92 @@
+name: GPU Compatibility
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'AUTHORS.md'
+      - 'CITATION.bib'
+      - 'CONTRIBUTING.md'
+      - 'LICENSE.md'
+      - 'NEWS.md'
+      - 'README.md'
+      - '.zenodo.json'
+      - '.github/workflows/benchmark.yml'
+      - '.github/workflows/CompatHelper.yml'
+      - '.github/workflows/TagBot.yml'
+      - 'benchmark/**'
+      - 'docs/**'
+      - 'utils/**'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
+    name: ${{ matrix.os }} - ${{ matrix.arch }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - version: '1.10'
+            os: ubuntu-latest
+            arch: x64
+          - version: '1.10'
+            os: windows-latest
+            arch: x64
+          - version: '1.10'
+            os: macOS-latest
+            arch: x64
+          - version: '1.10'
+            os: ubuntu-latest
+            arch: x86
+          - version: '1.10'
+            os: windows-latest
+            arch: x86
+          # Skip macOS x86 for now as Julia 1.10 binary support for macOS on x86 is unavailable
```

Comment:
We should not need so many OS, do we?

Reply:
> We should not need so many OS

Why?

Reply:
I'm now using macOS for profiling since CUDA.jl has some issues with profiling on Windows so I just assumed that CUDA still supports macOS (like on some older Macs).

Reply:
Review again please 😊